src/library/scala/collection/IterableOnce.scala

```diff
@@ -783,12 +783,7 @@ trait IterableOnceOps[+A, +CC[_], +C] extends Any { this: IterableOnce[A] =>
   def reduceRight[B >: A](op: (A, B) => B): B = this match {
     case seq: IndexedSeq[A @unchecked] if seq.length > 0 => foldr[A, B](seq, op)
     case _ if knownSize == 0 => throw new UnsupportedOperationException("empty.reduceRight")
-    case _ =>
-      try reversed.reduceLeft[B]((x, y) => op(y, x))    // reduceLeftIterator
-      catch {
-        case e: UnsupportedOperationException if e.getMessage == "empty.reduceLeft" =>
-          throw new UnsupportedOperationException("empty.reduceRight")
-      }
+    case _ => reversed.reduceLeft[B]((x, y) => op(y, x)) // reduceLeftIterator
```

Comment:
This is how it was before https://github.com/scala/scala/pull/10211.

The alternative would be to call `reduceLeftIterator` if `knwonSize == -1`, as it's done in `min` / `max`.